### PR TITLE
Reuse scripts to build the toolchain with one command

### PR DESCRIPTION
**Pull request by eukarpov**
Add toolchain build with single command to CI.
Fix params on cleaning ccache

### CODE REVIEW

**Comment by Blackhex** [comment]
Please note this PR https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/pull/137.

File: .github/scripts/build.sh [hunk 1]
```diff
@@ -20,6 +20,10 @@ if [ "$RUN_BOOTSTRAP" = 1 ]; then
     .github/scripts/install-dependencies.sh
 fi
 
+if [ "$UPDATE_SOURCE" = 1 ]; then
+    .github/scripts/update_source.sh
```
Comment (Blackhex) [inline .github/scripts/build.sh:24]: I'd use plural (`update_sources.sh`).

File: .github/scripts/build.sh [hunk 2]
```diff
@@ -20,6 +20,10 @@ if [ "$RUN_BOOTSTRAP" = 1 ]; then
     .github/scripts/install-dependencies.sh
 fi
 
+if [ "$UPDATE_SOURCE" = 1 ]; then
```
Comment (Blackhex) [inline .github/scripts/build.sh:23]: Why defining a new variable to enable the source update is needed? From my perspective using `RUN_BOOTSTRAP` for it is better.

File: .github/scripts/build.sh [hunk 3]
```diff
@@ -73,7 +77,7 @@ if [[ "$PLATFORM" =~ cygwin ]]; then
 fi
 
 if [ "$CCACHE" = 1 ] ; then
-  ccache -svv
+  ccache -s
```
Comment (Blackhex) [inline .github/scripts/build.sh:80]: Why was this line changed?

**Review by Blackhex — APPROVED** [review]
I made just minor comments, otherwise looks good.

File: .github/scripts/build.sh (hunk 2 above)
Comment (eukarpov) [reply .github/scripts/build.sh:23]: I would expect bootstrap should be needed less frequently and usually just once.
Update can happen every day to sync to the latest changes.

File: .github/scripts/build.sh (hunk 3 above)
Comment (eukarpov) [reply .github/scripts/build.sh:80]: It was failing. For me it looked like a typo.

File: .github/scripts/update_source.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+source `dirname ${BASH_SOURCE[0]}`/config.sh
+
+echo "::group::Update binutils, gcc and mingw sources"
+    [ ! -d "$SOURCE_PATH" ] && mkdir -p "$SOURCE_PATH"
```
[the remark this answers is not in the document]
Comment (eukarpov) [reply .github/scripts/update_source.sh:6]: It was added for clarity, however it can be refactored.

File: .github/scripts/build.sh (hunk 1 above)
Comment (eukarpov) [reply .github/scripts/build.sh:24]: also plural for UPDATE_SOURCE?

Comment (Blackhex) [reply .github/scripts/build.sh:24]: It would make sense to have both cosistent.

File: .github/scripts/build.sh (hunk 3 above)
Comment (Blackhex) [reply .github/scripts/build.sh:80]: Should be valid and should give the most verbose output https://ccache.dev/manual/4.10.html

Comment (eukarpov) [reply .github/scripts/build.sh:80]: why is it "vv"?

Comment (eukarpov) [reply .github/scripts/build.sh:80]: I have "ccache version 3.7.7" locally in WSL. It does not support -v also.
